File: 2020/2020W23/preppindataw23.R
library(dplyr)
library(readxl)
library(tidyr)

xlsx <- "F:/Data/Quiz Results.xlsx"

check_answer <- function(myans,correct) { 
  mapply(function(r,c) sum(r==c), strsplit(myans,','), strsplit(correct,',')) }

ans <- read_excel(xlsx, sheet = "Participant Answers") 
correct <- read_excel(xlsx, sheet = "Correct Answers") 

final <- ans %>%
  pivot_longer(., cols=matches('^(Round)'), names_to='Round', values_to='Result') %>%
  merge(correct, by='Round') %>%
  mutate('Score' = check_answer(`Result`,`Answers`)) %>%
  pivot_wider(., id_cols='Name', names_from='Round', values_from='Score') %>%
  mutate('Total Score' = `Round1`+`Round2`+`Round3`+`Round4`+`Round5`,
         'Position' = dense_rank(desc(as.numeric(`Total Score`)))) %>%
  select(c('Position','Name', 'Round1', 'Round2', 'Round3', 'Round4', 'Round5', 'Total Score'))

View(final)

File: 2020/2020W23/PreppindataW23.py
import pandas as pd

def check_answer(myans, correct):
    ok = [r for r,a in zip(myans, correct) if r==a]
    return len(list(filter((',').__ne__,ok)))

xlsx = pd.ExcelFile("F:/Data/Quiz Results.xlsx")

ans = pd.read_excel(xlsx,'Participant Answers')
correct = pd.read_excel(xlsx,'Correct Answers')

final = ans.melt(id_vars='Name',var_name='Round', value_name='Result')
final = final.merge(correct, on='Round')
final['Score'] = final.apply(lambda x: check_answer(x['Result'],x['Answers']),axis=1)

final = final.pivot_table(index='Name', values='Score', columns='Round').reset_index()
final['Total Score'] = final['Round1'] + final['Round2'] + final['Round3'] + final['Round4'] + final['Round5'] 
final['Position'] = final['Total Score'].rank(ascending=False,method='dense').astype(int)
final = final[['Position','Name', 'Round1', 'Round2', 'Round3', 'Round4', 'Round5', 'Total Score']]
